Software/Development/SemesterProject/App/dutyCycle.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * dutyCycle.c
+ *
+ *  Created on: 12 mar. 2019
+ *      Author: Aitor Teran
+ */
+//
+//###########################################################################
+
+//###########################################################################
+
+
+
+//
+// Includes
+//
+#include "Include/dutyCycle.h"
+#include "../System/Include/PWMConfig.h"
+
+
+//
+// Quasi-global variables definition
+//
+static struct dutyCycleTag
+{
+    DutyCycle legA;
+    DutyCycle legB;
+    DutyCycle legC;
+}dutyCycle;
+
+
+/* CONSIDER REVISION to reduce computing load.
+ * Function for internal use.
+ * map function performs a linear regression.
+ * Inputs are a value with its corresponding lower and upper limits
+ * as well as the lower and upper limits of the variable to return.
+ */
+Uint16 mapDuty(float x, float x_min, float x_max, Uint16 y_min, Uint16 y_max)
+{
+    return y_min + (Uint16)((float)(y_max-y_min)*(x-x_min)/(x_max-x_min));
+}
+
+/*
+ * Function for internal use.
+ * The duty cycle is checked in case it lies outside of boundaries.
+ * Later the struct containing the duty information gets updated.
+ */
+void fillDutyStruct(float duty, int leg)
```

Comment:
the argument leg in this case would be of type 'SysMgrState' which is probably wrong, but that is the name defined in the typedef oh the .h

Reply:
True, corrected.